sql-plugin/src/main/spark311/scala/org/apache/spark/sql/hive/rapids/shims/GpuSaveAsHiveFile.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "311"}
+{"spark": "312"}
+{"spark": "313"}
+{"spark": "320"}
+{"spark": "321"}
+{"spark": "321cdh"}
+{"spark": "321db"}
+{"spark": "322"}
+{"spark": "323"}
+{"spark": "330"}
+{"spark": "330cdh"}
+{"spark": "330db"}
+{"spark": "331"}
+{"spark": "332"}
+{"spark": "333"}
+{"spark": "340"}
+spark-rapids-shim-json-lines ***/
+package org.apache.spark.sql.hive.rapids.shims
+
+import com.nvidia.spark.rapids.{ColumnarFileFormat, GpuDataWritingCommand}
+import org.apache.hadoop.conf.Configuration
+
+import org.apache.spark.internal.io.FileCommitProtocol
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.catalog.BucketSpec
+import org.apache.spark.sql.catalyst.catalog.CatalogTypes.TablePartitionSpec
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.execution.SparkPlan
+import org.apache.spark.sql.execution.datasources.FileFormatWriter
+import org.apache.spark.sql.hive.execution.SaveAsHiveFile
+import org.apache.spark.sql.rapids.GpuFileFormatWriter
+
+// Base trait from which all hive insert statement physical execution extends.
+private[hive] trait GpuSaveAsHiveFile extends GpuDataWritingCommand with SaveAsHiveFile {
+
+  // TODO(future): Examine compressions options.
+  // - Apache Spark 3.1-3 has code to examine Hadoop compression settings
+  //   (and takes a FileSinkDesc instead of FileFormat).
+  // - Apache Spark 3.4 has removed all that logic.
+  // - GPU Hive text writer does not support compression for output.
```

Comment:
@andygrove Is the reason this is not unshimmed this TODO? cc @gerashegalov

Reply:
I think @mythrocks worked on this originally, so it would be good to get his opinion. I am going to go ahead and unshim for now though.